### PR TITLE
[GEOT-6842] : Add WPS 2.0 Support

### PR DESCRIPTION
**Pull request by eropartz**
JIRA : https://osgeo-org.atlassian.net/browse/GEOT-6842

Add Xml Support for WPS 20

<Include a few sentences describing the overall goals for this Pull Request>

# Checklist

Reviewing is a process done by project maintainers, **mostly on a volunteer basis** (thus limited in time). We need to keep the review overhead as small as possible, and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [X ] Confirm you have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md)
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [ X] Make sure the first PR targets the `main` branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.
- [ X] The changes are not causing two modules to share the same Java packages (to avoid [Java 9+ split package](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed) issues)
- [x] The changes are not breaking the build in downstream projects using SNAPSHOT dependencies, GeoWebCache and GeoServer (there is an automatic PR check verifying this, check this when it turns green).

The following are required only for core and extension modules (they are welcomed, but not required, for unsupported modules):

- [x] There is an issue in [Jira](https://osgeo-org.atlassian.net/projects/GEOT) describing the bug/task/new feature (a notable exemptions is, changes not visible to end users). The ticket is for the GeoTools project, if the issue was found elsewhere it's a good practice to link to the origin ticket/issue.
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ ] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message(s) must be in the form "[GEOT-XYZW] Title of the Jira ticket"
- [x] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [ ] Documentation has been updated accordingly.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.

### CODE REVIEW

**Comment by aaime** [comment]
This is a large code donation, cannot be considered without a CLA. 
I need to understand if the CLA is simply in progress, or there is anything blocking you to send one. If it's the latter, then we can avoid the review effort.

**Comment by eropartz** [comment]
> 
> 
> This is a large code donation, cannot be considered without a CLA.
> I need to understand if the CLA is simply in progress, or there is anything blocking you to send one. If it's the latter, then we can avoid the review effort.

Hi, The signin of the CLA is in progress, I wiil send you the document ASAP

**Comment by eropartz** [comment]
> 
> 
> > This is a large code donation, cannot be considered without a CLA.
> > I need to understand if the CLA is simply in progress, or there is anything blocking you to send one. If it's the latter, then we can avoid the review effort.
> 
> Hi, The signin of the CLA is in progress, I wiil send you the document ASAP

CLA sended

File: modules/extension/xsd/xsd-wps/src/main/java/org/geotools/wps/v2_0/bindings/ExecuteRequestTypeBinding.java [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+     * @generated modifiable
+     */
+    @Override
+    public Class getType() {
+        return ExecuteRequestType.class;
+    }
+
+    @Override
+    protected void setProperty(EObject eObject, String property, Object value, boolean lax) {
+        Object overridedValue = value;
```
Comment (aaime) [inline modules/extension/xsd/xsd-wps/src/main/java/org/geotools/wps/v2_0/bindings/ExecuteRequestTypeBinding.java:249]:
```suggestion
        Object overridenValue = value;
```

File: modules/extension/xsd/xsd-wps/src/main/java/org/geotools/wps/v2_0/bindings/DataDescriptionTypeBinding.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+     *
+     * <!-- begin-user-doc -->
+     * <!-- end-user-doc -->
+     *
+     * @generated modifiable
+     */
+    @Override
+    public Object parse(ElementInstance instance, Node node, Object value) throws Exception {
+
+        // TODO: implement and remove call to super
```
Comment (aaime) [inline modules/extension/xsd/xsd-wps/src/main/java/org/geotools/wps/v2_0/bindings/DataDescriptionTypeBinding.java:84]: Either remove this method or add the missing implementation?

File: modules/extension/xsd/xsd-wps/src/main/java/org/geotools/wps/v2_0/bindings/ComplexDataTypeBinding.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+     * <!-- begin-user-doc -->
+     * <!-- end-user-doc -->
+     *
+     * @generated modifiable
+     */
+    @Override
+    public Object parse(ElementInstance instance, Node node, Object value) throws Exception {
+
+        // TODO: implement and remove call to super
+        return super.parse(instance, node, value);
```
Comment (aaime) [inline modules/extension/xsd/xsd-wps/src/main/java/org/geotools/wps/v2_0/bindings/ComplexDataTypeBinding.java:102]: Either remove this method or add the missing implementation?

Generally speaking, I'm wondering why the implementation is completely separate from the WPS 1.0 one. 
In other services, when the two protocols are very similar, the bindings do share code, and sometimes they are a straight extension of each other, by subclassing (e.g., see SLD 1.1 vs SLD 1.0, WFS 1.1 vs WFS 1.0)

File: modules/extension/xsd/xsd-wps/src/main/java/org/geotools/wps/v2_0/bindings/DataInputTypeBinding.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+ *  		&lt;/attribute&gt;
+ *
+ *  	&lt;/complexType&gt;
+ *
+ *   </code>
+ * </pre>
+ *
+ * @generated
+ */
+public class DataInputTypeBinding extends AbstractComplexEMFBinding {
```
Comment (aaime) [inline modules/extension/xsd/xsd-wps/src/main/java/org/geotools/wps/v2_0/bindings/DataInputTypeBinding.java:79]: Most of these classes are adding nothing but noise, they could be replaced by an instance of ComplexEMFBinding.
See the WPS 1.0 configuraton and bindings, only a few bindings are implemneted as classes, all of the others are just instantiated in the configuration.

**Review by aaime — CHANGES_REQUESTED** [review]

File: modules/extension/xsd/xsd-wps/src/main/java/org/geotools/wps/v2_0/bindings/DataInputTypeBinding.java (hunk 4 above)
Comment (eropartz) [reply modules/extension/xsd/xsd-wps/src/main/java/org/geotools/wps/v2_0/bindings/DataInputTypeBinding.java]: Class removed. Use org.geotools.xsd.ComplexEMFBinding in WPSConfiguration to reduce number of bindings class

File: modules/extension/xsd/xsd-wps/src/main/java/org/geotools/wps/v2_0/bindings/ComplexDataTypeBinding.java (hunk 3 above)
Comment (eropartz) [reply modules/extension/xsd/xsd-wps/src/main/java/org/geotools/wps/v2_0/bindings/ComplexDataTypeBinding.java]: Class removed. Use org.geotools.xsd.ComplexEMFBinding in WPSConfiguration to reduce number of bindings class

File: modules/extension/xsd/xsd-wps/pom.xml [hunk 5]
```diff
@@ -1,126 +1,139 @@
...
-                                                                                
-        The Geotools Project                                                    
-            http://www.geotools.org/                                            
-                                                                                
-        Version: $Id$              
-     ======================================================================= -->
-  <project xmlns="http://maven.apache.org/POM/4.0.0" 
-           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
-           xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 
+<!-- =======================================================================
```
[the remark this answers is not in the document]
Comment (eropartz) [reply modules/extension/xsd/xsd-wps/pom.xml]: changes reverted

**Comment by aaime** [comment]
CLA verified, changes look good, merging. Thanks!

**Pull request merged**